OutOfSchool/OutOfSchool.Encryption/Program.cs

```diff
@@ -0,0 +1,73 @@
+using Asp.Versioning;
+using Elastic.Apm.DiagnosticSource;
+using OutOfSchool.Encryption.Handlers;
+using OutOfSchool.Encryption.Services;
+using Serilog;
+using Serilog.Context;
+using Serilog.Exceptions;
+using Serilog.Exceptions.Core;
+
+var builder = WebApplication.CreateBuilder(args);
+
+builder.Configuration
+    .AddJsonFile("appsettings.jsonc", false, true)
+    .AddJsonFile($"appsettings.{builder.Environment.EnvironmentName}.jsonc", true, true);
+
+builder.Host.UseSerilog((ctx, lc) => lc
+    .ReadFrom.Configuration(ctx.Configuration)
+    .Enrich.WithExceptionDetails(new DestructuringOptionsBuilder()
+        .WithDefaultDestructurers()));
+
+GlobalLogContext.PushProperty("AppVersion", builder.Configuration.GetSection("AppDefaults:Version").Value);
+
+builder.Services.AddElasticApmForAspNetCore(new HttpDiagnosticsSubscriber());
+
+if (builder.Environment.IsDevelopment())
+{
+    builder.Services.AddSingleton<IEUSignOAuth2Service, DevEUSignOAuth2Service>();
+}
+else
+{
+    builder.Services.AddSingleton<IEUSignOAuth2Service, EUSignOAuth2Service>();
+}
+
+builder.Services.AddApiVersioning(options =>
+    {
+        options.DefaultApiVersion = new ApiVersion(1);
+        options.ReportApiVersions = true;
+        options.AssumeDefaultVersionWhenUnspecified = true;
+        options.ApiVersionReader = ApiVersionReader.Combine(new UrlSegmentApiVersionReader());
+    })
+    .AddApiExplorer(options =>
+    {
+        options.GroupNameFormat = "'v'V";
+        options.SubstituteApiVersionInUrl = true;
+    });
+
+var app = builder.Build();
+
+app.UseSerilogRequestLogging();
```

Comment:
Does that logger hide the sensitive data like certs or keys or... ?

Reply:
`UseSerilogRequestLogging()` never logs body from requests, only URLs